tests/integration/testlib/ovslib.py

```diff
@@ -100,6 +109,32 @@ def create(self):
             }
             libnmstate.apply(desired_state, verify_change=False)
 
+    def apply(self):
+        libnmstate.apply(self.state)
+
+    @property
+    def state(self):
+        return {
+            Interface.KEY: _set_ifaces_state(self._ifaces, InterfaceState.UP)
+        }
+
+
+def get_ovs_port_nm_device_name(iface_name):
+    """
+    Given an interface name, return the OVS port device/profile name.
+    This assumes NM as the backend.
+    """
+    nmdev = device.get_device_by_name(iface_name)
+    ovs_port_dev = ovs.get_port_by_slave(nmdev)
+    return ovs_port_dev.get_iface()
+
+
+def list_profiles_names(iface_name):
```

Comment:
This helper does not seem to be specific to OVS, so it should be somewhere else. In other tests we use something with "show only" in its name to get the profiles. Despite its name, this is actually returning device names as it seems. Why is this special? Shouldn't it just use the other method?

Reply:
> This helper does not seem to be specific to OVS, so it should be somewhere else.
> In other tests we use something with "show only" in its name to get the profiles. Despite its name, this is actually returning device names as it seems. Why is this special? Shouldn't it just use the other method?

`show*` returns existing devices while this one returns defined profiles (including inactive ones).
It does not really matter what it returns, it could just become an assert helper.
I was mainly concern about not exposing `nm` on the tests directly.

I will create an `nmlib` module and push it there.